client/memez/projects/memez/src/app/stores/entities/like.store.ts

```diff
@@ -1,54 +1,73 @@
+import {autorun, toJS}      from 'mobx';
 import {Injectable}         from '@angular/core';
 import {action, observable} from 'mobx-angular';
-import {ILike}              from '../../types/Entities/ILike';
+import {ILike}              from '../../../../../../../../shared/types/Entities/ILike';
 import {RootStore}          from '../root.store';
-import {MOCK_LIKES}         from '../../../../../../../../shared/mock/MOCK_LIKES';
-import {IPost}              from '../../types/Entities/IPost';
-import * as dayjs           from 'dayjs';
-import {autorun}            from 'mobx';
-import {MatDialog}              from '@angular/material/dialog';
-import {LikeDialogBoxComponent} from '../../components/like-dialog-box/like-dialog-box.component';
+import {IPost}              from '../../../../../../../../shared/types/Entities/IPost';
 
 
 @Injectable({providedIn: 'root'})
 
 export class LikeStore {
   @observable likes: ILike[] = [];
-  USE_MOCK: boolean          = true;
+
+  /*USE_MOCK: boolean = false;*/
 
   constructor(
     public root: RootStore) {
     window['likeStore'] = this;
     this.root.likeStore = this;
+    autorun(() => {
+
+    })
+
 
   }
 
+
   @action
-  async getLikes(): Promise<ILike[]> {
-    if (this.USE_MOCK) {
-      return this.likes = MOCK_LIKES;
+  async createLike(post: IPost) {
+    const likeInput = {
+      user_id: this.root.log.currentUser._id,
+      post_id: post._id
     }
-    return this.likes = await this.root.likeAdapter.getLikes();
+    return await this.root.likeAdapter.createLike(likeInput)
```

Comment:
remember what I told you about returning an await inside an async function? The place where the function is used should await the promise to resolve because async keyword always returns a promise it will have to be awaited anyways